processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -0,0 +1,210 @@
+	if err != nil {
+		return false, fmt.Errorf("isEks() error retrieving auth configmap: %w", err)
+	}
+
+	return awsAuth != "", nil
+}
+
+// isK8s checks if the current environment is running in a Kubernetes environment
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// fileExists checks if a file with a given filename exists.
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	return err == nil && !info.IsDir()
+}
+
+// fetchString executes an HTTP request with a given HTTP Method and URL string.
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
+	}
+
+	// Set HTTP request header with authentication credentials
+	authHeader, err := getK8sCredHeader()
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	// Get certificate
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file with path %s", k8sCertPath)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	// Set HTTP request timeout and add certificate
+	client := &http.Client{
+		Timeout: timeoutMillis * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	response, err := client.Do(request)
+	if err != nil || response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to execute HTTP request with method=%s, URL=%s, Status Code=%d: %w", httpMethod, URL, response.StatusCode, err)
+	}
+
+	// Retrieve response body from HTTP request
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response from HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
+	}
+
+	return string(body), nil
+}
+
+// getK8sCredHeader retrieves the kubernetes credential information.
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("getK8sCredHeader() error: cannot read file with path %s", k8sTokenPath)
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+// getClusterName retrieves the clusterName resource attribute
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: %w", err)
+	}
+
+	// parse JSON object returned from HTTP request
+	var respmap map[string]json.RawMessage
+	err = json.Unmarshal([]byte(resp), &respmap)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: cannot parse JSON: %w", err)
+	}
+	var d data
+	err = json.Unmarshal(respmap["data"], &d)
```

Comment:
Is this a JSON inside JSON?

Reply:
Sure appears that way. :-P I still have end to end testing to do.